src/main/java/dev/waterdog/waterdogpe/ProxyServer.java

```diff
@@ -473,6 +475,14 @@ public void setForcedHostHandler(IForcedHostHandler forcedHostHandler) {
         this.forcedHostHandler = forcedHostHandler;
     }
 
+    public IMetricsHandler getMetricsHandler() {
+        return metricsHandler;
+    }
+
+    public void setMetricsHandler(IMetricsHandler metricsHandler) {
+        this.metricsHandler = metricsHandler;
```

Comment:
Since we do not perform any null checks when accessing metrics, this should rather thrown an exception here than on other places.

Reply:
good point

Reply:
Fixed in https://github.com/WaterdogPE/WaterdogPE/commit/0e6572444ff044e07473dfa155636b96ca0eadeb